compiler/damlc/daml-compiler/src/DA/Daml/Compiler/Upgrade.hs

```diff
@@ -381,20 +392,47 @@ generateSrcFromLf thisPkgId pkgMap m = noLoc mod
     convType =
         \case
             LF.TVar tyVarName ->
-                HsTyVar NoExt NotPromoted $
-                mkRdrName $ LF.unTypeVarName tyVarName
-            LF.TCon LF.Qualified {..} ->
-                HsTyVar NoExt NotPromoted $
-                noLoc $
-                mkOrig
-                    (mkModule
-                         (getUnitId qualPackage)
-                         (mkModuleName $
-                          T.unpack $ LF.moduleNameString qualModule))
-                    (mkOccName varName $
-                     T.unpack $ T.intercalate "." $ LF.unTypeConName qualObject)
+                HsTyVar NoExt NotPromoted $ mkRdrName $ LF.unTypeVarName tyVarName
+            LF.TCon LF.Qualified {..}
```

Comment:
I'd prefer a big `case LF.unTypeConName qualObject of` and then cases `[name]`, `n@[_, _]`, etc.

Reply:
done.